core/installer/base/deps_version_parser.lua

```diff
@@ -1,5 +1,5 @@
 -- 解析单个版本字符串到版本对象
-function parse_single_version(version_str)
+function parse_single_version_range(version_str)
     local version = {}
     if version_str == "any" or version_str == "*" then
```

Comment:
local Version = {}
Version.__index = Version

function Version:new(str)
      local self = setmetatable({},Version}
      self:parse(str)
      return self
end

function Version:parse(str)
   local pattern = (...)
   local major,minor,patch,prerelease,build = str:match(pattern)
   self.major = tonumber(major) or 0
   self.minor = tonumber(minor) or 0
   self.patch = tonumber(patch) or 0
   self.prerelease = tonumber(prerelease) or ""
   self.build = tonumber(build) or "" 
end

function Version:compare(other)
    if self.major ~= other.major then
        return self.major - other.major
    end
    if self.minor ~= other.minor then
        return self.minor - other.minor
    end
    if self.patch ~= other.patch then
        return self.patch - other.patch
    end

    if self.prerelease ~= other.prerelease then
        if self.prerelease == "" then
            return 1
        elseif other.prerelease == "" then
            return -1
        else
            return self.prerelease < other.prerelease and -1 or 1
        end
    end

    return 0
end

function Version:__lt(other)
    return self:compare(other) < 0
end

function Version:__le(other)
    return self:compare(other) <= 0
end

function Version:__eq(other)
    return self:compare(other) == 0
end

function Version:__gt(other)
    return self:compare(other) > 0
end

function Version:__ge(other)
    return self:compare(other) >= 0
end